### DISCUSSION TITLE
Customize table query

### DISCUSSION THREAD

**Discussion by Sirloko** [filamentphp/filament#8859] [Ideas]
Problem description
In filament 2, i could customize the table query like this

protected function getTableQuery(): Builder
{
switch ($user_role = auth()->user()->roles->first()->name) {
case 'Staff':
return Model::query()->where('status', 0);
break;
case 'Vendor':
return Model::query()->where('status', 3);
break;
}
}

but after migrating to filament 3, i cant seem to find a way to do so. perhaps i'm missing something

**Comment by leandrocfe** [comment]
https://filamentphp.com/docs/3.x/panels/resources/listing-records/#customizing-the-table-eloquent-query

**Reply by Sirloko** [reply]
Thanks for the resource.
